src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -571,6 +572,25 @@ private String encodePath(String path) {
         return Arrays.stream(path.split("/")).map(segment -> percentEncode(segment)).collect(Collectors.joining("/"));
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PackageURL that = (PackageURL) o;
```

Comment:
Nit: Maybe rename `that` to `other` as "that" visually looks similar to "this" (yes, I get the pun), and "other" would also better match the "o" parameter name.

Reply:
Done